src/http-api.ts

```diff
@@ -0,0 +1,72 @@
+import type { RequestContext } from './lib/json-rpc-router.js';
+
+import cors from 'cors';
+import express from 'express';
+import getRawBody from 'raw-body';
+import ContentType from 'content-type';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { jsonRpcApi } from './json-rpc-api.js';
+import { createJsonRpcErrorResponse, JsonRpcErrorCodes } from './lib/json-rpc.js';
+
+export const httpApi = express();
+
+httpApi.use(cors());
+
+httpApi.get('/health', (req, res) => {
+  // return 200 ok
+  return res.json({ ok: true });
+});
+
+httpApi.post('/', async (req, res) => {
+  let dwnRequest;
+
+  try {
+    // the json rpc request payload is provided in the http request body _or_ as the value of
+    // the 'dwn-request' request header.
+    dwnRequest = req.headers['dwn-request'] ?? await getRawBody(req, { encoding: true });
+  } catch(e) {
+    const reply = createJsonRpcErrorResponse(
+      uuidv4(), JsonRpcErrorCodes.BadRequest, 'failed to read request.');
```

Comment:
i dont think this is valid, as the [spec](https://www.jsonrpc.org/specification#response_object) states
> If there was an error in detecting the id in the Request object (e.g. Parse error/Invalid Request), it MUST be Null.

as such we probably should use `null` instead of `uuidv4()`